From 05ba237af582c1ca3780e5fe06ab3320494efe52 Mon Sep 17 00:00:00 2001
From: Lukas Slebodnik <lslebodn@redhat.com>
Date: Sat, 13 Oct 2018 16:22:13 +0000
Subject: [PATCH 80/83] CONFIGURE: Add minimal required version for p11-kit

There are few functions which were added in upstream p11-kit 0.23.3.
And there are compilation failures with older versions.

src/p11_child/p11_child_openssl.c: In function 'get_pkcs11_uri':
src/p11_child/p11_child_openssl.c:87:12: error: implicit declaration of function 'p11_kit_uri_get_slot_info' [-Werror=implicit-function-declaration]
     memcpy(p11_kit_uri_get_slot_info(uri), slot_info, sizeof(CK_SLOT_INFO));
            ^

src/p11_child/p11_child_openssl.c:88:5: error: implicit declaration of function 'p11_kit_uri_set_slot_id' [-Werror=implicit-function-declaration]
     p11_kit_uri_set_slot_id(uri, slot_id);
     ^

src/p11_child/p11_child_openssl.c: In function 'do_card':
src/p11_child/p11_child_openssl.c:767:35: error: implicit declaration of function 'p11_kit_uri_get_slot_id' [-Werror=implicit-function-declaration]
                     uri_slot_id = p11_kit_uri_get_slot_id(uri);
                                   ^
src/p11_child/p11_child_openssl.c:770:32: error: implicit declaration of function 'p11_kit_uri_match_slot_info' [-Werror=implicit-function-declaration]
                             || p11_kit_uri_match_slot_info(uri, &info) != 1) {
                                ^
Merges: https://pagure.io/SSSD/sssd/pull-request/3852

Reviewed-by: Sumit Bose <sbose@redhat.com>
---
 src/external/p11-kit.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/external/p11-kit.m4 b/src/external/p11-kit.m4
index a959f43..eb0474f 100644
--- a/src/external/p11-kit.m4
+++ b/src/external/p11-kit.m4
@@ -1,4 +1,4 @@
 AC_SUBST(P11_KIT_CFLAGS)
 AC_SUBST(P11_KIT_LIBS)
 
-PKG_CHECK_MODULES([P11_KIT], [p11-kit-1])
+PKG_CHECK_MODULES([P11_KIT], [p11-kit-1 >= 0.23.3])
-- 
2.9.5

